File: src/AltEvent.Core/Models/Event.cs
using AltEvent.Core.Utils;
using System.Text.Json.Serialization;

namespace AltEvent.Core.Models
{
    public class Event : BaseEntity
    {
        public long Id { get; set; }

        public long CompanyId { get; set; }

        public long CreatorId { get; set; }

        public string Name { get; set; }

        public string? Description { get; set; }

        public bool AllDay { get; set; }

        [JsonConverter(typeof(DateOnlyJsonConverter))]
        public DateOnly StartDate { get; set; }

        [JsonConverter(typeof(DateOnlyJsonConverter))]
        public DateOnly EndDate { get; set; }

        [JsonConverter(typeof(TimeOnlyJsonConverter))]
        public TimeOnly? StartTime { get; set; }

        [JsonConverter(typeof(TimeOnlyJsonConverter))]
        public TimeOnly? EndTime { get; set; }

        public string Location { get; set; }

        public int Capacity { get; set; }

        [JsonIgnore]
        public bool IsDeleted { get; set; }

        //

        public Company Company { get; set; }

        public User Creator { get; set; }

        //

        public ICollection<User> Hosts { get; set; }

        public ICollection<Reservation> Reservations { get; set; }
    }
}

File: src/AltEvent.WebApp/src/core/models/event.ts
import { Company } from "./company";
import { Reservation } from "./reservation";
import { User } from "./user";

export class Event {
    public id!: number;
    public companyId!: number;
    public creatorId!: number;
    public name!: string;
    public description?: string;
    public allDay!: boolean;
    public startDate!: Date;
    public endDate!: Date;
    public startTime?: Date;
    public endTime?: Date;
    public location!: string;
    //public locationId: number;
    public capacity!: number;
    public createdAt!: Date;
    public updatedAt!: Date;

    public company!: Company;
    public creator!: User;
    //public location!: Location;

    public hosts!: User[];
    public reservations!: Reservation[];
}
